pytube/extract.py

```diff
@@ -396,3 +398,48 @@ def apply_descrambler(stream_data: Dict, key: str) -> None:
         ]
 
     logger.debug("applying descrambler")
+
+
+def initial_data(watch_html: str) -> str:
+    """
+    Extracts the initial_data json from the watch_html page.
+    This is used for extraction of copyright notices located at the bottom
+    of the video description
+    @param watch_html: Html of the watch page
+    @return:
+    """
+    initial_data_pattern = r"window\[['\"]ytInitialData['\"]]\s*=\s*([^\n]+)"
+    try:
+        match = regex_search(initial_data_pattern, watch_html, 1)
+    except RegexMatchError:
+        return "{}"
+    else:
+        return match[:-1]
+
+
+def metadata(initial_data) -> Optional[YouTubeMetadata]:
+    """Get the metadata for the video.
+
+    :rtype: YouTubeMetadata
+    """
+    try:
+        metadata_rows: List = initial_data["contents"]["twoColumnWatchNextResults"][
+            "results"]["results"]["contents"][1]["videoSecondaryInfoRenderer"][
+            "metadataRowContainer"]["metadataRowContainerRenderer"]["rows"]
+    except (KeyError, IndexError):
+        # If there's an exception accessing this data, it probably doesn't exist.
+        return YouTubeMetadata([])
+
+    # Rows appear to only have "metadataRowRenderer" or "metadataRowHeaderRenderer"
+    #  and we only care about the former, so we filter the others
+    metadata_rows = list(
+        filter(
+            lambda x: "metadataRowRenderer" in x.keys(),
+            metadata_rows
+        )
+    )
+    # We then access the metadataRowRenderer key in each element
+    #  and build a metadata object from this new list
+    metadata_rows = [x["metadataRowRenderer"] for x in metadata_rows]
```

Comment:
[x["metadataRowRenderer"] for x in metadata_rows if "metadataRowRenderer" in x.keys()]

prbably should dump "metadataRowRenderer" somewhere in top of class and explain how we got it / if it's just youtube magic constant

Reply:
I actually explain this on 444-445. The metadata_rows are all objects with a single key (either "metadataRowRenderer" or "metadataRowHeaderRenderer"), and the value of that key is the actual metadata.